pkg/services/ngalert/schedule/schedule.go

```diff
@@ -78,17 +81,11 @@ func (sch *schedule) definitionRoutine(grafanaCtx context.Context, key models.Al
 						"key", key, "attempt", attempt, "now", ctx.now, "duration", end.Sub(start), "error", err)
 					return err
 				}
-				for _, r := range results {
-					sch.log.Info("alert definition result", "title", alertDefinition.Title, "key", key, "attempt", attempt, "now", ctx.now, "duration", end.Sub(start), "instance", r.Instance, "state", r.State.String())
-					cmd := models.SaveAlertInstanceCommand{DefinitionOrgID: key.OrgID, DefinitionUID: key.DefinitionUID, State: models.InstanceStateType(r.State.String()), Labels: models.InstanceLabels(r.Instance), LastEvalTime: ctx.now}
-					err := sch.store.SaveAlertInstance(&cmd)
-					if err != nil {
-						sch.log.Error("failed saving alert instance", "title", alertDefinition.Title, "key", key, "attempt", attempt, "now", ctx.now, "instance", r.Instance, "state", r.State.String(), "error", err)
-					}
-				}
+
 				transitionedStates := stateTracker.ProcessEvalResults(key.DefinitionUID, results, condition)
+				sch.saveAlertStates(transitionedStates)
```

Comment:
should rename this since it is now a state for all evaluated results, to make persistence more robust